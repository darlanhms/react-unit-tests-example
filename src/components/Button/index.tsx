import styles from "./styles.module.css";

interface ButtonProps {
  variant?: "primary" | "secondary";
}

const Button: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  variant = "primary",
  children,
}) => {
  return (
    <button className={[styles[variant], styles.base].join(" ")}>
      {children}
    </button>
  );
};

export default Button;
