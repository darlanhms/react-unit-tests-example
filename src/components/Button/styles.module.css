.base {
  color: white;
  padding: 8px 25px;
  border: none;
  border-radius: 5px;
}

.primary {
    background-color: red;
}

.secondary {
    background-color: blue;
}
