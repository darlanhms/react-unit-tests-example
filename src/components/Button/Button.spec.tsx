import { render, screen } from "@testing-library/react";
import Button from ".";

const variants = ["primary", "secondary"] as const;

describe("Button", () => {
  test.each(variants)("renders variant %s correctly", (variant) => {
    const { container } = render(<Button variant={variant}>button</Button>);

    const button = screen.getByText("button");

    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
