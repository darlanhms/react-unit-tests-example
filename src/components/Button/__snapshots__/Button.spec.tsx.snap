// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders variant primary correctly 1`] = `
<div>
  <button
    class=" "
  >
    button
  </button>
</div>
`;

exports[`Button renders variant secondary correctly 1`] = `
<div>
  <button
    class=" "
  >
    button
  </button>
</div>
`;
